### PR TITLE
tests: add patch, put, int64, cap.letter tests

### PR DESCRIPTION
**Pull request by georgiyekkert**
tests for:
1)int 64 field - b/190201970
2)a field that starts with cap letter - b/184008102
3) put, patch

### CODE REVIEW

File: compute/apiv1/smoke_test.go [hunk 1]
```diff
@@ -396,3 +452,125 @@ func TestPaginationMapResponseMaxRes(t *testing.T) {
...
+		Project: projectId,
+		Image:   name,
+	})
+
+	fetched, err := c.Get(ctx,
+		&computepb.GetImageRequest{
+			Project: projectId,
+			Image:   name,
+		})
+	if !reflect.DeepEqual(fetched.GetLicenseCodes(), codes) {
```
Comment (noahdietz) [inline compute/apiv1/smoke_test.go:510]: Please use `go-cmp` instead. https://pkg.go.dev/github.com/google/go-cmp/cmp

```go
if diff := cmp.Diff(fetched.GetLicenseCodes(), codes); diff != "" {
    t.Fatalf("got(-),want(+):\n%s", diff)
}
```

File: compute/apiv1/smoke_test.go [hunk 2]
```diff
@@ -396,3 +452,125 @@ func TestPaginationMapResponseMaxRes(t *testing.T) {
...
+			Project:   projectId,
+			Operation: insert.Proto().GetName(),
+		})
+	if err != nil {
+		t.Error(err)
+	}
+	defer func(c *FirewallsClient, ctx context.Context, req *computepb.DeleteFirewallRequest) {
+		_, err := c.Delete(ctx, req)
+		if err != nil {
+		}
```
Comment (noahdietz) [inline compute/apiv1/smoke_test.go:564]: Is this necessary if there it is never handled?

File: compute/apiv1/smoke_test.go [hunk 3]
```diff
@@ -396,3 +452,125 @@ func TestPaginationMapResponseMaxRes(t *testing.T) {
...
+			Operation: insert.Proto().GetName(),
+		})
+	if err != nil {
+		t.Error(err)
+	}
+
+	defer func(c *ImagesClient, ctx context.Context, req *computepb.DeleteImageRequest) {
+		_, err := c.Delete(ctx, req)
+		if err != nil {
+		}
```
Comment (noahdietz) [inline compute/apiv1/smoke_test.go:499]: Is this necessary?

File: compute/apiv1/smoke_test.go [hunk 4]
```diff
@@ -396,3 +452,125 @@ func TestPaginationMapResponseMaxRes(t *testing.T) {
...
+	}(c, ctx, &computepb.DeleteFirewallRequest{
+		Project:  projectId,
+		Firewall: name,
+	})
+	fetched, err := c.Get(ctx, &computepb.GetFirewallRequest{
+		Project:  projectId,
+		Firewall: name,
+	})
+	if !reflect.DeepEqual(fetched.GetAllowed(), allowed) {
```
Comment (noahdietz) [inline compute/apiv1/smoke_test.go:573]: Please use `go-cmp`.
```go
if diff := cmp.Diff(fetched.GetLicenseCodes(), codes, cmp.Comparer(proto.Equal)); diff != "" {
    t.Fatalf("got(-),want(+):\n%s", diff)
}
```

File: compute/apiv1/smoke_test.go [hunk 5]
```diff
@@ -396,3 +452,125 @@ func TestPaginationMapResponseMaxRes(t *testing.T) {
...
+	_, err = globalClient.Wait(ctx,
+		&computepb.WaitGlobalOperationRequest{
+			Project:   projectId,
+			Operation: insert.Proto().GetName(),
+		})
+	if err != nil {
+		t.Error(err)
+	}
+
+	defer func(c *ImagesClient, ctx context.Context, req *computepb.DeleteImageRequest) {
```
Comment (noahdietz) [inline compute/apiv1/smoke_test.go:496]: Are the `func` params necessary? They should be captured in the closure if they are defined in the lines preceding this `defer`

Comment (georgiyekkert) [reply compute/apiv1/smoke_test.go]: done! didn't expect that params would be captured.

File: compute/apiv1/smoke_test.go (hunk 3 above)
Comment (georgiyekkert) [reply compute/apiv1/smoke_test.go]: added t.error()

File: compute/apiv1/smoke_test.go (hunk 1 above)
Comment (georgiyekkert) [reply compute/apiv1/smoke_test.go]: done

File: compute/apiv1/smoke_test.go (hunk 2 above)
Comment (georgiyekkert) [reply compute/apiv1/smoke_test.go]: now handled

File: compute/apiv1/smoke_test.go (hunk 4 above)
Comment (georgiyekkert) [reply compute/apiv1/smoke_test.go]: done

File: compute/apiv1/smoke_test.go [hunk 6]
```diff
@@ -107,7 +107,7 @@ func TestCreateGetPutPatchListInstance(t *testing.T) {
 	if get.GetDescription() != "тест" {
 		t.Fatal(fmt.Sprintf("expected instance description: %s, got: %s", "тест", get.GetDescription()))
 	}
-	if get.GetShieldedInstanceConfig().GetEnableSecureBoot() != false {
+	if secureBootEnabled := get.GetShieldedInstanceConfig().GetEnableSecureBoot(); secureBootEnabled {
 		t.Fatal(fmt.Sprintf("expected instance secure boot: %t, got: %t", false, get.GetShieldedInstanceConfig().GetEnableSecureBoot()))
```
Comment (noahdietz) [inline compute/apiv1/smoke_test.go:111]: `Fatalf` instead of `Fatal(fmt.Sprintf`

File: compute/apiv1/smoke_test.go [hunk 7]
```diff
@@ -157,10 +157,10 @@ func TestCreateGetPutPatchListInstance(t *testing.T) {
 		t.Error(err)
 	}
 	if fetched.GetDescription() != "updated" {
-		t.Fatal(fmt.Sprintf("expected instance description: %s, got: %s", "updated", get.GetDescription()))
+		t.Fatal(fmt.Sprintf("expected instance description: %s, got: %s", "updated", fetched.GetDescription()))
```
Comment (noahdietz) [inline compute/apiv1/smoke_test.go:160]: `Fatalf` instead of `Fatal(fmt.Sprintf`

File: compute/apiv1/smoke_test.go [hunk 8]
```diff
@@ -157,10 +157,10 @@ func TestCreateGetPutPatchListInstance(t *testing.T) {
...
 	}
-	if fetched.GetShieldedInstanceConfig().GetEnableSecureBoot() != true {
-		t.Fatal(fmt.Sprintf("expected instance secure boot: %t, got: %t", true, get.GetShieldedInstanceConfig().GetEnableSecureBoot()))
+	if secureBootEnabled := fetched.GetShieldedInstanceConfig().GetEnableSecureBoot(); !secureBootEnabled {
+		t.Fatal(fmt.Sprintf("expected instance secure boot: %t, got: %t", true, secureBootEnabled))
```
Comment (noahdietz) [inline compute/apiv1/smoke_test.go:163]: `Fatalf` instead of `Fatal(fmt.Sprintf`

File: compute/apiv1/smoke_test.go [hunk 9]
```diff
@@ -492,23 +492,24 @@ func TestTypeInt64(t *testing.T) {
 	if err != nil {
 		t.Error(err)
 	}
-
-	defer func(c *ImagesClient, ctx context.Context, req *computepb.DeleteImageRequest) {
-		_, err := c.Delete(ctx, req)
-		if err != nil {
-		}
-	}(c, ctx, &computepb.DeleteImageRequest{
+	delReq := &computepb.DeleteImageRequest{
```
Comment (noahdietz) [inline compute/apiv1/smoke_test.go:495]: Nit: This doesn't need to define this outside of the `defer`, it can be inlined at the function call. My previous mention of that was for regarding the other data, `name` and `projectId`.

File: compute/apiv1/smoke_test.go [hunk 10]
```diff
@@ -558,19 +559,21 @@ func TestCapitalLetter(t *testing.T) {
 	if err != nil {
 		t.Error(err)
 	}
-	defer func(c *FirewallsClient, ctx context.Context, req *computepb.DeleteFirewallRequest) {
-		_, err := c.Delete(ctx, req)
-		if err != nil {
-		}
-	}(c, ctx, &computepb.DeleteFirewallRequest{
+	delReq := &computepb.DeleteFirewallRequest{
```
Comment (noahdietz) [inline compute/apiv1/smoke_test.go:562]: Nit: This doesn't need to define this outside of the `defer`, it can be inlined at the function call. My previous mention of that was for regarding the other data, `name` and `projectId`.

**Review by noahdietz — COMMENTED** [review]
Can you run `goimports` please? That's why CI failed.

```
go get golang.org/x/tools/cmd/goimports
cd compute/apiv1
goimports -w .
```

File: compute/apiv1/smoke_test.go (hunk 7 above)
Comment (georgiyekkert) [reply compute/apiv1/smoke_test.go:161]: done

File: compute/apiv1/smoke_test.go (hunk 8 above)
Comment (georgiyekkert) [reply compute/apiv1/smoke_test.go:164]: done

File: compute/apiv1/smoke_test.go (hunk 6 above)
Comment (georgiyekkert) [reply compute/apiv1/smoke_test.go]: done

File: compute/apiv1/smoke_test.go (hunk 9 above)
Comment (georgiyekkert) [reply compute/apiv1/smoke_test.go]: done

File: compute/apiv1/smoke_test.go (hunk 10 above)
Comment (georgiyekkert) [reply compute/apiv1/smoke_test.go]: done

File: go.mod [hunk 11]
```diff
@@ -10,6 +10,7 @@ require (
 	github.com/googleapis/gax-go/v2 v2.0.5
 	go.opencensus.io v0.23.0
 	golang.org/x/oauth2 v0.0.0-20210805134026-6f1e6394065a
+	golang.org/x/tools v0.1.5 // indirect
```
Comment (noahdietz) [inline go.mod:13]: Please revert this change. It was because of the `go get` I asked you to run.

File: go.sum [hunk 12]
```diff
@@ -212,6 +212,7 @@ golang.org/x/mod v0.2.0/go.mod h1:s0Qsj1ACt9ePp/hMypM3fl4fZqREWJwdYDEqhRiZZUA=
 golang.org/x/mod v0.3.0/go.mod h1:s0Qsj1ACt9ePp/hMypM3fl4fZqREWJwdYDEqhRiZZUA=
 golang.org/x/mod v0.4.0/go.mod h1:s0Qsj1ACt9ePp/hMypM3fl4fZqREWJwdYDEqhRiZZUA=
 golang.org/x/mod v0.4.1/go.mod h1:s0Qsj1ACt9ePp/hMypM3fl4fZqREWJwdYDEqhRiZZUA=
+golang.org/x/mod v0.4.2 h1:Gz96sIWK3OalVv/I/qNygP42zyoKp3xptRVCWRFEBvo=
```
Comment (noahdietz) [inline go.sum:215]: Please revert this change. It was because of the `go get` I asked you to run.

File: go.mod (hunk 11 above)
Comment (georgiyekkert) [reply go.mod]: reverted

File: go.sum (hunk 12 above)
Comment (georgiyekkert) [reply go.sum]: reverted

File: compute/apiv1/smoke_test.go [hunk 13]
```diff
@@ -396,3 +453,128 @@ func TestPaginationMapResponseMaxRes(t *testing.T) {
...
+			&computepb.DeleteImageRequest{
+				Project: projectId,
+				Image:   name,
+			})
+		if err != nil {
+			t.Error(err)
+		}
+	}()
+
+	fetched, err := c.Get(ctx,
```
Comment (noahdietz) [inline compute/apiv1/smoke_test.go:507]: check the error here after this.

File: compute/apiv1/smoke_test.go [hunk 14]
```diff
@@ -396,3 +453,128 @@ func TestPaginationMapResponseMaxRes(t *testing.T) {
...
+		_, err := c.Delete(ctx,
+			&computepb.DeleteFirewallRequest{
+				Project:  projectId,
+				Firewall: name,
+			})
+		if err != nil {
+			t.Error(err)
+		}
+	}()
+	fetched, err := c.Get(ctx, &computepb.GetFirewallRequest{
```
Comment (noahdietz) [inline compute/apiv1/smoke_test.go:573]: check the error here after this.

File: compute/apiv1/smoke_test.go (hunk 13 above)
Comment (georgiyekkert) [reply compute/apiv1/smoke_test.go:507]: done

File: compute/apiv1/smoke_test.go (hunk 14 above)
Comment (georgiyekkert) [reply compute/apiv1/smoke_test.go:576]: done

**Review by noahdietz — APPROVED** [review]
